components/filter/filter.js

```diff
@@ -350,14 +410,44 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 		this._stopPropagation(e);
 	}
 
+	_handleSearch(e) {
+		e.stopPropagation();
+		const dimension = !this._activeDimensionKey ? this._dimensions[0] : this._dimensions.find(dimension => dimension.key === this._activeDimensionKey);
+		const dimensionNode = !this._activeDimensionKey ? this._dimensionNodes[0] : this._dimensionNodes.find(node => node.key === this._activeDimensionKey);
+		dimension.searchValue = e.detail.value;
+		dimensionNode.searchValue = e.detail.value;
+
+		if (dimension.searchType === 'event') {
```

Comment:
Do we have a use case for this type of asynchronous search right away? Just wondering if it's worth implementing all this extra logic where the simple search should hopefully meet a lot of use cases. Or would it be for places that might have a lot of values and require some form of paging?

Reply:
So My Courses uses HM calls to search.  It doesn't do paging, but was maybe setting itself up to one day do so. Or maybe it searches deeper than just the department name (to the description)?  I've never had to dig into that api, so that's just a guess

Reply:
Right, so possibly a combination of searching multiple fields (which may or may not even be rendered) and paging.